protocol/protocol_test.go

```diff
@@ -301,22 +301,32 @@ func TestChaining(t *testing.T) {
 
 	for _, ver := range allVersions {
 		t.Run(ver.String(), func(t *testing.T) {
-			nonce1, _, _, err := CreateRequest([]Version{ver}, rand.Reader, nil, rootPublicKeyA)
+			_, _, req1Bytes, err := CreateRequest([]Version{ver}, rand.Reader, nil, rootPublicKeyA)
```

Comment:
I suppose it's a little redundant to have `CreateRequest` return the nonce, given that it can be parsed from the request bytes, but no harm in keeping it for convenience.

Reply:
Hmm, if we dropped this then the client would have to parse the nonce from the request in order to verify the response.

A nicer alternative might be to return an actual `*Request` that the client encodes itself. I suggest we address this in a future PR since it's not related to this one.